aggregator/pkg/middlewares/anonymous_auth_middleware.go

```diff
@@ -23,24 +55,72 @@ func (m *AnonymousAuthMiddleware) Intercept(ctx context.Context, req any, info *
 		return handler(ctx, req)
 	}
 
-	if ip, ok := tryGetIP(ctx); ok {
+	if ip, ok := m.tryGetIP(ctx); ok {
 		ctx = auth.ToContext(ctx, auth.CreateCallerIdentity(ip, true))
 	}
 
 	return handler(ctx, req)
 }
 
-func tryGetIP(ctx context.Context) (string, bool) {
+func (m *AnonymousAuthMiddleware) isTrustedProxy(ipStr string) bool {
+	// Extract IP from "ip:port" format
+	host, _, err := net.SplitHostPort(ipStr)
+	if err != nil {
+		host = ipStr // No port, use as-is
+	}
+
+	ip := net.ParseIP(host)
+	if ip == nil {
+		return false
+	}
+
+	// Normalize IPv4 to 4-byte format for consistent comparison.
+	// net.ParseIP returns a 16-byte representation, but parseCIDROrIP
+	// stores IPv4 trusted proxies as 4-byte IPNets. Both must use the
+	// same format for IPNet.Contains to match correctly.
+	if ip4 := ip.To4(); ip4 != nil {
+		ip = ip4
+	}
+
+	for _, trusted := range m.trustedProxies {
+		if trusted.Contains(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *AnonymousAuthMiddleware) tryGetIP(ctx context.Context) (string, bool) {
+	// If no trusted proxies are configured, anonymous auth is disabled
```

Comment:
I think you can still allow it, the issue is just that you can't trust the headers unless a trusted proxy is enabled. So it should roughly be like:

```
peer = getPeer(ctx)

if isTrustedProxy(peer):
	# use headers to get client for rate limiting
else:
	# rate limit by peer
```

I think we'd want to support deployments without a reverse proxy like in staging envs etc.